libs/SalesforceSDK/src/com/salesforce/androidsdk/developer/support/notifications/local/ShowDeveloperSupportNotifier.kt

```diff
@@ -0,0 +1,204 @@
+package com.salesforce.androidsdk.developer.support.notifications.local
+
+import android.Manifest.permission.POST_NOTIFICATIONS
+import android.app.Activity
+import android.app.Application
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.NotificationManager.IMPORTANCE_DEFAULT
+import android.app.PendingIntent
+import android.app.PendingIntent.FLAG_IMMUTABLE
+import android.content.Context.MODE_PRIVATE
+import android.content.Context.NOTIFICATION_SERVICE
+import android.content.Intent
+import android.content.Intent.FLAG_ACTIVITY_SINGLE_TOP
+import android.content.pm.PackageManager.PERMISSION_GRANTED
+import android.os.Build.VERSION.SDK_INT
+import android.os.Build.VERSION_CODES.O
+import android.os.Build.VERSION_CODES.TIRAMISU
+import androidx.annotation.RequiresApi
+import androidx.core.app.ActivityCompat
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationCompat.PRIORITY_DEFAULT
+import androidx.core.app.NotificationManagerCompat
+import androidx.core.content.edit
+import com.salesforce.androidsdk.R.drawable.sf__salesforce_logo
+import com.salesforce.androidsdk.R.string.sf__notifications_local_show_dev_support_action_title
+import com.salesforce.androidsdk.R.string.sf__notifications_local_show_dev_support_content
+import com.salesforce.androidsdk.R.string.sf__notifications_local_show_dev_support_text
+import com.salesforce.androidsdk.R.string.sf__notifications_local_show_dev_support_title
+import com.salesforce.androidsdk.developer.support.activities.ShowDeveloperSupportActivity
+import com.salesforce.androidsdk.developer.support.notifications.local.ShowDeveloperSupportNotifier.Companion.NotificationId.SHOW_DEVELOPER_SUPPORT
+
+/**
+ * Provides the "Show Developer Support" local notification, including
+ * notification permissions and set up.
+ */
+internal class ShowDeveloperSupportNotifier {
```

Comment:
This class is a container for the functions which manage the notifications and permissions.  Most of it should seem pretty boilerplate.